src/components/TradeAssetSearch/components/SearchTermAssetList.tsx

```diff
@@ -57,6 +65,14 @@ export const SearchTermAssetList = ({
     return assets.filter(asset => asset.chainId === activeChainId)
   }, [activeChainId, allowWalletUnsupportedAssets, assets, walletConnectedChainIds])
 
+  const handleAssetClick = useCallback(
+    (asset: Asset) => {
+      onAssetClick(asset)
+      fetchAndUpsertCustomTokenMarketData(asset.assetId)
+    },
+    [fetchAndUpsertCustomTokenMarketData, onAssetClick],
+  )
+
   const customAssets: Asset[] = useMemo(
     () =>
       customTokens
```

Comment:
nitpick - `(customTokens ?? []).map` to reduce the need for ternary and then can just return on `!metadata` in the map so we only need to `filter(isSome)` once at the end. probably never going to be that many custom tokens to notice, but just making note

Reply:
https://github.com/shapeshift/web/pull/7370/commits/9f7c5f1860557b4f526baab4c81aa1cdd6029341